reconcile/openshift_saas_deploy_trigger_base.py

```diff
@@ -63,12 +67,158 @@ def setup(options):
+            oc_map,
+            already_triggered,
+            settings,
+            state_update_method,
+            integration,
+            integration_version)
+
+    else:
+        error = True
+        logging.error(
+            f'[{saas_file_name}] unsupported provider: ' +
+            f'{provider_name}'
+        )
+
+    return error
+
+
+def _trigger_jenkins(spec,
+                     dry_run,
+                     jenkins_map,
+                     already_triggered,
+                     settings,
+                     state_update_method):
+    # TODO: Convert these into a dataclass.
+    saas_file_name = spec['saas_file_name']
+    env_name = spec['env_name']
+    pipelines_provider = spec['pipelines_provider']
+
+    instance_name = pipelines_provider['instance']['name']
+    job_name = get_openshift_saas_deploy_job_name(
+        saas_file_name, env_name, settings)
+
+    error = False
+    to_trigger = _register_trigger(job_name, already_triggered)
+    if to_trigger:
+        logging.info(['trigger_job', instance_name, job_name])
+        if not dry_run:
+            jenkins = jenkins_map[instance_name]
+            try:
+                jenkins.trigger_job(job_name)
+                state_update_method(spec)
+            except Exception as e:
+                error = True
+                logging.error(
+                    f"could not trigger job {job_name} " +
+                    f"in {instance_name}. details: {str(e)}"
+                )
+
+    return error
+
+
+def _trigger_tekton(spec,
+                    dry_run,
+                    oc_map,
+                    already_triggered,
+                    settings,
+                    state_update_method,
+                    integration,
+                    integration_version):
+    # TODO: Convert these into a dataclass.
+    saas_file_name = spec['saas_file_name']
+    env_name = spec['env_name']
+    timeout = spec['timeout']
+    pipelines_provider = spec['pipelines_provider']
+
+    tkn_namespace_info = pipelines_provider['namespace']
+    tkn_namespace_name = tkn_namespace_info['name']
+    tkn_cluster_name = tkn_namespace_info['cluster']['name']
+    tkn_trigger_resource, tkn_name = _construct_tekton_trigger_resource(
+        saas_file_name,
+        env_name,
+        timeout,
+        settings,
+        integration,
+        integration_version
+    )
+
+    error = False
+    to_trigger = _register_trigger(tkn_name, already_triggered)
+    if to_trigger:
+        try:
+            osb.apply(dry_run=dry_run,
+                      oc_map=oc_map,
+                      cluster=tkn_cluster_name,
+                      namespace=tkn_namespace_name,
+                      resource_type=tkn_trigger_resource.kind,
+                      resource=tkn_trigger_resource,
+                      wait_for_namespace=False)
+            if not dry_run:
+                state_update_method(spec)
+        except Exception as e:
+            error = True
```

Comment:
We have a stylistic disagreement here. If you let the exception go, the caller can try to implement some automatic remediation on its own. It is also far less code to drop this exception handling and let the top level do the final handling and reporting. Also, IMHO and FWIIW such top level is `run_integration` and not `run`.

Reply:
One of the things we decided a while back was that even if specific actions fail, we still want to reconcile as much as possible, therefore the agreement here was to continue the execution but report an error (so we can be aware of this).

I agree with what you said the other day about not doing sys.exit in the integrations (in the run methods) but rather in the run_integration, which makes sense, but I think this example here is a bit different. It's an interesting conversation to have because maybe I'm not visualizing your proposal for this particular situation, but since this is the current agreement, I would suggest to not change this in this PR, but rather come up with a proposal and an action plan, if we agree to.